src/hint_processor/builtin_hint_processor/dict_manager.rs

```diff
@@ -88,9 +90,19 @@ impl DictManager {
             ));
         };
 
+        let mut floored_initial = HashMap::new();
+        for (key, val) in initial_dict.into_iter() {
+            let floored_val = if val.is_negative() {
+                val.mod_floor(&vm.prime)
+            } else {
+                val
+            };
+            floored_initial.insert(key, floored_val);
+        }
+
```

Comment:
Can we use map for this?

Reply:
Resolved: https://github.com/lambdaclass/cairo-rs/pull/614/commits/99220f47aac9f24a5dd0f20b533d8a512f66b6e9